shell/browser/api/atom_api_desktop_capturer.cc

```diff
@@ -94,43 +94,23 @@ void DesktopCapturer::StartHandling(bool capture_window,
           content::DesktopMediaID::TYPE_WINDOW,
           content::desktop_capture::CreateWindowCapturer());
       window_capturer_->SetThumbnailSize(thumbnail_size);
-      window_capturer_->AddObserver(this);
-      window_capturer_->StartUpdating();
+      window_capturer_->Update(base::BindOnce(
+          &DesktopCapturer::UpdateSourcesList, weak_ptr_factory_.GetWeakPtr(),
+          window_capturer_.get()));
```

Comment:
I'm not confident enough in my C++ to be sure this is the optimal way to pass these pointers; please double check.